Microsoft.Research/Contracts/System/System.Collections.Specialized.INotifyCollectionChanged.cs

```diff
@@ -0,0 +1,359 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace System.Collections.Specialized
+{
+#if !SILVERLIGHT && !NETFRAMEWORK_3_5 // => types defined in WindowsBase!
+    // Summary:
+    //     Describes the action that caused a System.Collections.Specialized.INotifyCollectionChanged.CollectionChanged
+    //     event.
+    public enum NotifyCollectionChangedAction
+    {
+        // Summary:
+        //     One or more items were added to the collection.
+        Add = 0,
+        //
+        // Summary:
+        //     One or more items were removed from the collection.
+        Remove = 1,
+        //
+        // Summary:
+        //     One or more items were replaced in the collection.
+        Replace = 2,
+        //
+        // Summary:
+        //     One or more items were moved within the collection.
+        Move = 3,
+        //
+        // Summary:
+        //     The content of the collection changed dramatically.
+        Reset = 4,
+    }
+
+
+    // Summary:
+    //     Provides data for the System.Collections.Specialized.INotifyCollectionChanged.CollectionChanged
+    //     event.
+    public class NotifyCollectionChangedEventArgs : EventArgs
+    {
+        // Summary:
+        //     Initializes a new instance of the System.Collections.Specialized.NotifyCollectionChangedEventArgs
+        //     class that describes a System.Collections.Specialized.NotifyCollectionChangedAction.Reset
+        //     change.
+        //
+        // Parameters:
+        //   action:
+        //     The action that caused the event. This must be set to System.Collections.Specialized.NotifyCollectionChangedAction.Reset.
+        public NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction action) { }
+        //
+        // Summary:
+        //     Initializes a new instance of the System.Collections.Specialized.NotifyCollectionChangedEventArgs
+        //     class that describes a multi-item change.
+        //
+        // Parameters:
+        //   action:
+        //     The action that caused the event. This can be set to System.Collections.Specialized.NotifyCollectionChangedAction.Reset,
+        //     System.Collections.Specialized.NotifyCollectionChangedAction.Add, or System.Collections.Specialized.NotifyCollectionChangedAction.Remove.
+        //
+        //   changedItems:
+        //     The items that are affected by the change.
+        public NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction action, IList changedItems) { }
```

Comment:
I'm fine with that as long as you create an issue. If the issue is narrow-scoped, it might make it easier for a new contributor to come along and add them. An example:

> ## Add contracts for NotifyCollectionChangedEventArgs
>
> A stub contract class for `NotifyCollectionChangedEventArgs` was added in #84. The contracts for methods in this class still need to be added.

Then we can mark the issue with the **up for grabs** label. :+1: